src/app/services/xml-parsers/bibliography-parser.service.ts

```diff
@@ -0,0 +1,147 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { map, shareReplay } from 'rxjs/operators';
+import { EditionDataService } from '../edition-data.service';
+
+function getStringFromTags(tagLabel: string, cit: Element) {
+  let res = '';
+  Array.from(cit.getElementsByTagName(tagLabel)).map(element => res += element.textContent.replace(/\s+/g, ' ') + ' ');
+
+  return res.trim();
+}
+
+@Injectable({
+  providedIn: 'root',
+})
+export class BibliographyParserService {
+  private findedCits = { complete: 0, total: 0 };
+
+  constructor(
+    private edition: EditionDataService,
+  ) {
+  }
+
+  public getBibliographicCitations(): Observable<CitationsFeature> {
+    const bibliographicCitations: BibliographicCitation[] = [];
+
+    return this.edition.parsedEditionSource$.pipe(
+      map(responses => {
+        Array.from(responses.getElementsByTagName('bibl')).map(citation => {
+          if (citation.getElementsByTagName('author').length === 0 &&
```

Comment:
What if there is a only a date or a title and not all three?

Reply:
Please, try to handle all possible encoding situation, by using both examples from TEI guidelines (https://www.tei-c.org/release/doc/tei-p5-doc/en/html/examples-listBibl.html, https://www.tei-c.org/release/doc/tei-p5-doc/en/html/examples-bibl.html and https://www.tei-c.org/release/doc/tei-p5-doc/en/html/examples-biblStruct.html) and other possible encoding situations